x/admin/keeper/msg_server.go

```diff
@@ -0,0 +1,80 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sagaxyz/saga-sdk/x/admin/types"
+)
+
+var _ types.MsgServer = &Keeper{}
+
+var ErrNotAuthorized = errors.New("not authorized")
+var ErrInvalidRequest = errors.New("invalid request")
+
+func (k Keeper) SetMetadata(
+	goCtx context.Context,
+	msg *types.MsgSetMetadata,
+) (*types.MsgSetMetadataResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if msg.Metadata == nil {
+		return nil, errorsmod.Wrap(ErrInvalidRequest, "metadata is nil")
+	}
+
+	params := k.GetParams(ctx)
+	isACLAdmin := params.Permissions.SetMetadata &&
+		k.aclKeeper != nil &&
+		k.aclKeeper.Admin(ctx, sdk.AccAddress(msg.Authority))
```

Comment:
i think it would be clearer to rename this method on the acl keeper to `IsAdmin`. Just `Admin` could also be interpreted as setting the value 🙏 
```suggestion
		k.aclKeeper.IsAdmin(ctx, sdk.AccAddress(msg.Authority))
```